presto-main/src/main/java/com/facebook/presto/sql/planner/PlanFragment.java

```diff
@@ -144,6 +144,11 @@ public boolean isLeaf()
         return remoteSourceNodes;
     }
 
+    public boolean hasDefaultOutput()
```

Comment:
Hmmm, it is something more general to me. In the future, maybe other operators will produce some non empty output for empty input (i.e. `defaultOutput`).

If not, do you have other proposal?